platform/core/src/services/MeasurementService/MeasurementService.ts

```diff
@@ -549,26 +554,27 @@ class MeasurementService extends PubSubService {
       );
     }
 
+    const oldMeasurement = this.measurements[internalUID];
+
     const newMeasurement = {
+      ...oldMeasurement,
       ...measurement,
       modifiedTimestamp: Math.floor(Date.now() / 1000),
       uid: internalUID,
     };
 
-    if (this.measurements[internalUID]) {
+    if (oldMeasurement) {
       // TODO: Ultimately, each annotation should have a selected flag right from the soure.
       // For now, it is just added in OHIF here and in setMeasurementSelected.
-      newMeasurement.selected = this.measurements[internalUID].selected;
       this.measurements[internalUID] = newMeasurement;
       if (isUpdate) {
```

Comment:
Why this is removed? I'm pretty sure @joe added that recently for selection

Reply:
It is implicit in the way that the new measurement is created, so was redundant.  It wasn't the only thing that was getting updated, which is why I changed how the update happens.